lib/ansible/modules/cloud/amazon/aws_elasticbeanstalk_app.py

```diff
@@ -0,0 +1,261 @@
+        "Versions": [
+            "1.0.0",
+            "1.0.1"
+        ]
+    }
+apps:
+    description: list of beanstalk applications
+    returned: when state == list
+    type: list
+    sample: [
+        {
+            "ApplicationName": "app1",
+            "ConfigurationTemplates": [],
+            "DateCreated": "2016-12-28T14:50:03.185000+00:00",
+            "DateUpdated": "2016-12-28T14:50:03.185000+00:00",
+            "Description": "description"
+        },
+        {
+            "ApplicationName": "app2",
+            "ConfigurationTemplates": [],
+            "DateCreated": "2016-12-28T14:50:03.185000+00:00",
+            "DateUpdated": "2016-12-28T14:50:03.185000+00:00",
+            "Description": "description"
+        }
+    ]
+output:
+    description: message indicating what change will occur
+    returned: in check mode
+    type: string
+    sample: App is up-to-date
+'''
+
+
+try:
+    import boto3
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import boto3_conn, ec2_argument_spec, get_aws_connection_info
+
+
+def describe_app(ebs, app_name):
+    apps = list_apps(ebs, app_name)
+
+    return None if len(apps) != 1 else apps[0]
+
+
+def list_apps(ebs, app_name):
+    if app_name is not None:
+        apps = ebs.describe_applications(ApplicationNames=[app_name])
+    else:
+        apps = ebs.describe_applications()
+
+    return apps.get("Applications", [])
+
+
+def check_app(ebs, app, module):
+    app_name = module.params['app_name']
+    description = module.params['description']
+    state = module.params['state']
+    terminate_by_force = module.params['terminate_by_force']
+
+    result = {}
+
+    if state == 'present' and app is None:
+        result = dict(changed=True, output="App would be created")
+    elif state == 'present' and app.get("Description", None) != description:
+        result = dict(changed=True, output="App would be updated", app=app)
+    elif state == 'present' and app.get("Description", None) == description:
+        result = dict(changed=False, output="App is up-to-date", app=app)
+    elif state == 'absent' and app is None:
+        result = dict(changed=False, output="App does not exist")
+    elif state == 'absent' and app is not None:
+        result = dict(changed=True, output="App will be deleted", app=app)
+    elif state == 'absent' and app is not None and terminate_by_force is True:
+        result = dict(changed=True, output="Running environments terminated before the App will be deleted", app=app)
+
+    module.exit_json(**result)
+
+
+def filter_empty(**kwargs):
+    return {k: v for k, v in iter(kwargs.items()) if v}
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+
+    argument_spec.update(
+        dict(
+            app_name=dict(aliases=['name'], type='str', required=False),
+            description=dict(type='str', required=False),
```

Comment:
`type='str'` and `required=False` are the defaults - this can be `description=dict()`